app/utils/invoke-action.js

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable no-extra-semi */
+import { assert } from '@ember/debug';
+import { get } from '@ember/object';
+
+function invokeAction(object, actionName, ...args) {
+  assert('The first argument passed to invokeAction must be an object', typeof object === 'object');
+
+  let action;
+  if (typeof actionName === 'string') {
+    action = get(object, actionName);
+  } else if (typeof actionName === 'function') {
+    action = actionName;
+  } else {
+    assert('The second argument passed to invokeAction must be a string as actionName or a function', false);
+  }
+
+  if (typeof action === 'string') {
+    object.sendAction(actionName, ...args);
+  } else if (typeof action === 'function') {
+    return action(...args);
+  }
+}
+
+export { invokeAction }
```

Comment:
yo no metería esta abstracción. Lo único que hace es chequear que lo que venga sea un string o una función para elegir si hacer f() o this.sendAction('f'), y en tu código lo usás siempre como un string. ¿O hay otros lugares?

Reply:
Y si no existe no la invoca. Que al final era lo que sucedía con el componente que necesitamos, por lo que lo terminé sacando.